apps/dashboard/app/javascript/script_edit.js

```diff
@@ -148,38 +148,82 @@ function addInProgressField(event) {
   enableNewFieldButton();
 }
 
+function fixExcludeBasedOnSelect(selectElement) {
+  const excludeElementId = selectElement.dataset.excludeId;
+  const selectOptions = Array.from(selectElement.options);
+  const itemsToExclude = selectOptions.filter(opt => !opt.selected).map(opt => opt.text);
+  const excludeElement = document.getElementById(excludeElementId);
+  excludeElement.value = itemsToExclude.join(',');
+}
+
 function fixedFieldEnabled(checkbox, dataElement) {
   // Disable the element to avoid updates from the user
   dataElement.disabled = true;
   // As it is disabled, need to add a hidden field with the same name to send the fixed field value to the backend.
   const input = $('<input>').attr('type','hidden').attr('name', dataElement.name).attr('value', dataElement.value);
   $(checkbox).after(input);
+
+  if (dataElement.nodeName == 'SELECT') {
+    const selectOptions = Array.from(dataElement.options);
+    const selectedOption = selectOptions.filter(opt => opt.selected)[0];
+    const selectOptionsConfig = $(dataElement).parents('.editable-form-field').find('li.list-group-item').get();
+
+    selectOptionsConfig.forEach(configItemLi => {
+      const textContent = $(configItemLi).find('[data-select-value]')[0].textContent;
+      if (selectedOption.text == textContent) {
+        enableSelectOptionConfig(configItemLi, true);
+      } else {
+        disableSelectOptionConfig(configItemLi, true);
+      }
+    });
+  }
 }
 
 function toggleFixedField(event) {
   event.target.disabled = true;
-  const elementId = event.target.dataset.fixedToggler;
-  const dataElement = document.getElementById(elementId);
+  const targetId = event.target.dataset.fixedToggler;
+  const dataElement = document.getElementById(targetId);
   if (event.target.checked) {
     fixedFieldEnabled(event.target, dataElement)
   } else {
     dataElement.disabled = false;
     // Field enabled, remove the hidden field with the same name needed when disabled.
     $(`input[type=hidden][name="${dataElement.name}"]`).remove();
+
+    if (dataElement.nodeName == 'SELECT') {
+      fixExcludeBasedOnSelect(dataElement);
+      initSelect(dataElement);
+    }
   }
 
   event.target.disabled = false;
 }
 
+function disableSelectOptionConfig(liElement, addButtonDisabled = false) {
```

Comment:
The fact that these methods have a boolean to their interface is a code smell to me.  I know why you have it like that, but I would ask if there's some other way to do this without the second boolean argument.

Reply:
At a minimum, it should likely change names to `disableBothButtons` or similar. 

Both of these function always disable at least 1 button and something like `addButtonDisabled` does accurately describe what it does, but that's sort of after you realize/know that `disableSelectOptionConfig` _always_ disables the remove button and this flag is for the other button.

In any case, I think the name change makes it simpler: both these disable _a_ button, with the option to disable both without having to think hard about which one is doing what.

Reply:
I think this is my only sticking point with this PR. I would ask if you can see if there's a way to do it without booleans in interfaces.

If not, then that's OK and we can live with it, but I'd like to at least investigate an alternative at any rate..

Reply:
I have updated with other solution. Let me know what you think.

Reply:
I'm going to apologize for this upfront: I think the first solution is better. I know I kind of asked for the impossible - but at least the first solution had modifications co-located to each other and I would guess that's easier to reason about.

How about we go back to the original implementation with a name change. I'd guess that's the best we can do.

Reply:
OK - will do

Reply:
I settle for the new names: `enableAddOption`, `enableRemoveOption`.
The name `disableBothButtons` didn't quite represent what the function was doing and there is the other function that does the opposite. Although both functions need to disable buttons.

What do you think about the new names?
Other suggestions for both methods?